### PR TITLE
Add support for properties and property definitions

### PR DESCRIPTION
**Pull request by taimoorgit**
## Issues

https://github.com/OpsLevel/product/issues/8

Relies on https://github.com/OpsLevel/opslevel-go/pull/365

## Changelog

- [x] Update config version to 1.3.0 - supports properties
- [x] Add code and tests
- [x] Make a `changie` entry

## Tophatting

I used the example configs in docs and made the nginx pods so I could test easily.

```
6:19PM INF [random-nginx-3] No changes detected to fields - skipping update
6:19PM INF [random-nginx-3] All tags already assigned to service.
6:19PM INF [random-nginx-3] Successfully assigned property: prop{def='key1key2', value='{"key1": "val1", "key2": "val2"}'}
6:19PM INF [random-nginx-1] No changes detected to fields - skipping update
6:19PM INF [random-nginx-1] All tags already assigned to service.
6:19PM INF [random-nginx-1] Successfully assigned property: prop{def='color_picker', value='"green"'}
6:19PM INF [random-nginx-2] No changes detected to fields - skipping update
6:19PM INF [random-nginx-2] All tags already assigned to service.
6:19PM ERR [random-nginx-2] Failed assigning property: prop{def='color_picker', value='"turquoise"'} error="OpsLevel API Errors:\n\t- 'value' Value doesn't satisfy the schema: root value 'turquoise' is not one of: red, green, blue, yellow, purple\n"
6:19PM WRN unexpected happened: got service with alias 'self-hosted-random-nginx-4' but the result has no ID
6:19PM INF [random-nginx-4] Created new service
6:19PM INF [random-nginx-4] Assigned alias 'self-hosted-random-nginx-4'
6:19PM INF [random-nginx-4] Assigned tags: [{"key":"k8s_created","value":"2024-02-02T23:14:17Z"},{"key":"hello","value":"world"},{"key":"app.kubernetes.io/name","value":"proxy"}]
6:19PM ERR [random-nginx-4] Failed assigning property: prop{def='key1key2', value='"hello world"'} error="OpsLevel API Errors:\n\t- 'value' Value doesn't satisfy the schema: root of type string is not of type: hash\n"
```

### CODE REVIEW

File: src/common/reconciler.go [hunk 1]
```diff
@@ -361,3 +362,34 @@ func (r *ServiceReconciler) handleRepositories(service *opslevel.Service, regist
 		}
 	}
 }
+
+func (r *ServiceReconciler) handleProperties(service *opslevel.Service, registration opslevel_jq_parser.ServiceRegistration) {
+	for def, val := range registration.Properties {
+		definition := opslevel.NewIdentifier(def)
+		owner := opslevel.NewIdentifier(string(service.Id))
+		// TODO: hack to get around NewJSONInput not reading strings properly.
```
Comment (taimoorgit) [inline src/common/reconciler.go:370]: We need a separate function for parsing JSON from strings OR we need to add logic to `opslevel.NewJSONInput` to handle strings this way. Without this, strings are sent unquoted and the server outputs `unexpected token at`...

File: docs/example-opslevel-k8s.yaml [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+        excludes:
+          - .metadata.namespace == "kube-system"
+          - .metadata.annotations."opslevel.com/ignore"
+      opslevel:
+        name: .metadata.name
+        description: .metadata.annotations."opslevel.com/description"
+        owner: .metadata.annotations."opslevel.com/owner"
+        aliases:
+          - '"\(.metadata.namespace)-\(.metadata.name)"'
+        properties:
```
Comment (taimoorgit) [inline docs/example-opslevel-k8s.yaml:16]: Updated example from the docs with some properties, 1 string 1 object.

File: src/common/reconciler.go (hunk 1 above)
Comment (taimoorgit) [reply src/common/reconciler.go]: see https://github.com/OpsLevel/opslevel-go/pull/365

**Review by rocktavious — APPROVED** [review]
🚀  Great Work!

**Pull request merged**